File: DaS.ScriptLib/Game/Data Types/BossFightInfo.vb
Public Class BossFightInfo
    Public Name As String = "?BossRushBossName?"
    Public PlayerWarp As EntityLocation = EntityLocation.Zero
    Public World As Integer = 0
    Public Area As Integer = 0
    Public WarpID As Integer = -1
    Public PlayerAnim As Integer = 7410
    Public AdditionalFlags As Integer() = {}
    Public EventFlag As Integer = -1
    Public EntranceLua As String = ""

    Public Sub New(name As String)
        Me.Name = name
    End Sub

End Class

File: DaS.ScriptLib.CS/Game/Data/Structures/BossFightInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace DaS.ScriptLib.Game.Data.Structures
{
	public class BossFightInfo
	{
		public EventFlag.Boss SpawnFlag { get; }
		public string Name { get; }

		public Loc PlayerWarp = Loc.Zero;
		public int World = 0;
		public int Area = 0;
		public int WarpID = -1;
		public int PlayerAnim = 7410;
		public int[] AdditionalFlags = {
			
		};

		public string EntranceLua = "";
		public BossFightInfo(EventFlag.Boss spawnFlag, string name)
		{
			this.SpawnFlag = spawnFlag;
			_Name = name;
		}

	}
}
